### PR TITLE
Private/gokay/canvastilelayer2

### PR DESCRIPTION
**Pull request by gokaysatir**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [x] Code is properly formatted
- [x] All commits have Change-Id
- [ ] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: loleaflet/src/control/Control.Header.js [hunk 1]
```diff
@@ -585,26 +580,15 @@ L.Control.Header = L.Control.extend({
...
-				this._canvasWidth = value;
-			// console.log('Header._setCanvasSizeImpl: _canvasWidth' + this._canvasWidth);
-		}
-		else if (property === 'height') {
-			canvas.height = Math.floor(value * scale);
-			if (!isCorner)
-				this._canvasHeight = value;
-			// console.log('Header._setCanvasSizeImpl: _canvasHeight' + this._canvasHeight);
-		}
+		L.DomUtil.setStyle(container, property, value + 'px');
```
Comment (kendy) [inline loleaflet/src/control/Control.Header.js:579]: Previously the setStyle was not called in the !value case, now it is always, it is intended?

File: loleaflet/src/core/Util.js [hunk 2]
```diff
@@ -164,10 +164,6 @@ L.Util = {
 		if (!useExactDPR)
 			dpiScale = Math.ceil(dpiScale);
 
-		if (dpiScale == 1 && L.Browser.retina) {
```
Comment (kendy) [inline loleaflet/src/core/Util.js:167]: The removal here does not look like a cosmetic change, but a functional one?

File: loleaflet/src/layer/tile/CanvasTileLayer.js [hunk 3]
```diff
@@ -125,8 +113,8 @@ L.CanvasTilePainter = L.Class.extend({
 	_syncTileContainerSize: function () {
 		var tileContainer = this._layer._container;
 		if (tileContainer) {
-			tileContainer.style.width = this._width + 'px';
```
Comment (kendy) [inline loleaflet/src/layer/tile/CanvasTileLayer.js:128]: This, together with the _setCanvasSize (and whatever related) would be worth a separate commit, it is very non-obvious for code reading I fear...

File: loleaflet/src/layer/tile/CanvasTileLayer.js [hunk 4]
```diff
@@ -154,15 +143,14 @@ L.CanvasTilePainter = L.Class.extend({
...
-		    [viewBounds];
-		var canvasCorePx = new L.Point(this._pixWidth, this._pixHeight);
+		var paneBoundsList = splitPanesContext ? splitPanesContext.getPxBoundList(viewBounds): [viewBounds];
 
-		return { canvasSize: canvasCorePx,
-			 tileSize: tileSize,
-			 viewBounds: viewBounds,
-			 paneBoundsList: paneBoundsList };
+		return {
+			canvasSize: new L.Point(this._pixWidth, this._pixHeight),
```
Comment (kendy) [inline loleaflet/src/layer/tile/CanvasTileLayer.js]: The info that this is core pixels got lost; please either add a comment, or keep the local var.

File: loleaflet/src/layer/tile/CanvasTileLayer.js [hunk 5]
```diff
@@ -144,7 +132,8 @@ L.CanvasTilePainter = L.Class.extend({
 				this._canvasCtx.fillStyle = 'rgba(255, 0, 0, 0.5)';
 			else
 				this._canvasCtx.fillStyle = 'white';
-			this._canvasCtx.fillRect(0, 0, this._pixWidth, this._pixHeight);
+
+			this._canvasCtx.fillRect(0, 0, parseFloat(this._canvas.style.width.replace('px', '')), parseFloat(this._canvas.style.height.replace('px', '')));
```
Comment (kendy) [inline loleaflet/src/layer/tile/CanvasTileLayer.js]: Another one that should go together with the _setCanvasSize change I think?

File: loleaflet/src/layer/tile/CanvasTileLayer.js [hunk 6]
```diff
@@ -107,7 +107,7 @@ L.CanvasTilePainter = L.Class.extend({
 	},
 
 	canvasDPIScale: function () {
-		return parseInt(this._canvas.width) / this._width;
+		return parseInt(this._canvas.width) / parseFloat(this._canvas.style.width.replace('px', ''));
```
Comment (kendy) [inline loleaflet/src/layer/tile/CanvasTileLayer.js]: This is not the first time I see the "parseFloat(this._canvas.style.width.replace('px', ''))" - shouldn't we have an own method for this (and the height)?

File: loleaflet/src/control/Control.Header.js (hunk 1 above)
Comment (gokaysatir) [reply loleaflet/src/control/Control.Header.js:579]: Hello, yes it was intended, this function almost never called without value.

File: loleaflet/src/core/Util.js (hunk 2 above)
Comment (gokaysatir) [reply loleaflet/src/core/Util.js:167]: I thought that this code never gets hit. If L.Browser.retina  equals to "true", then dpiScale should not equal to 1.

**Review by gokaysatir — COMMENTED** [review]
canvasDPIScale function is removed in following comits. parseFloat on file CanvasTileLayer will not be there.

Yes we can merge SetCanvasSize and SyncTileContainerSize functions. Those changes remove _width and _height variables. But if we merge it (if it is possible)  then we can remove them.

line:
this._canvasCtx.fillRect(0, 0, this._pixWidth, this._pixHeight); 
should also be updated in following commits. But this line is in function "Clear", i think it should stay there?

I will keep the local variable inside _paintContext function.

**Comment by kendy** [comment]
image

With the patches applied, I see very thick lines in the rows header?

**Comment by gokaysatir** [comment]
Yes i they are a bit thick. I tried to adjust them by comparing them with the current working version. Then i made them look similar. But i can make them thinner with another patch of course.

**Comment by kendy** [comment]
> canvasDPIScale function is removed in following comits. parseFloat on file CanvasTileLayer will not be there.
> 
> Yes we can merge SetCanvasSize and SyncTileContainerSize functions. Those changes remove _width and _height variables. But if we merge it (if it is possible) then we can remove them.

I didn't mean to merge those two, I meant to separate the changes to those to a separate commit, to better understand the intention there :-)

**Comment by gokaysatir** [comment]
I see sorry :) Those ones were only cosmetic changes, i removed a local variable and pasted the code into its place. Since local variable was used only once. I thought that we don't need that local variable.
